src/main/java/sally/Task.java

```diff
@@ -0,0 +1,140 @@
+     */
+    public void unmark() {
+        toBeDone = true;
+    }
+
+    /**
+     * Returns the status icon indicating whether the task is done or not done.
+     *
+     * @return The status icon as a string.
+     */
+    public String getStatusIcon() {
+        return (toBeDone ? "[ ]" : "[X]");
+    }
+
+    /**
+     * Returns a string representation of the task formatted for storage in a file.
+     * If the task is an instance of Todo, Event, or Deadline, returns their specific string representations.
+     *
+     * @return The string representation of the task for storage.
+     */
+    public String toFileString() {
+        if (this instanceof Todo) {
+            Todo todo = (Todo) this;
+            return todo.toString();
+        } else if (this instanceof Event) {
+            Event event = (Event) this;
+            return event.toString();
+        } else if (this instanceof Deadline) {
+            Deadline deadline = (Deadline) this;
+            return deadline.toString();
+        } else {
+            return this.toString();
+        }
+    }
+
+    /**
+     * Converts a string representation of date and time to a LocalDateTime object.
+     *
+     * @param input The input string representing date and time.
+     * @return The LocalDateTime object parsed from the input string.
+     */
+    private static LocalDateTime convertToDateTime(String input) {
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
+        return LocalDateTime.parse(input, inputFormatter);
+    }
+
+    /**
+     * Converts a string representation of a task from a file to a Task object.
+     *
+     * @param fileString The string representation of the task from the file.
+     * @return The Task object created from the file string.
+     */
+    public static Task fromFileString(String fileString) {
+        String[] split = fileString.split("]\\[");
+        String splitOne = split[0]; // [D
+        String[] toGetTaskType = splitOne.split("\\[");
+        String taskType = toGetTaskType[1]; // D
+
+        String splitTwo = split[1]; // " ] soc (by:sun)"
+        String[] toGetStatusIcon = splitTwo.split("] ");
+        String statusIcon = toGetStatusIcon[0]; // " "
+        boolean isDone = false;
+
+        if (statusIcon.equals(" ")) {
+            isDone = false;
+        } else {
+            isDone = true;
+        }
+
+        String taskDescription = toGetStatusIcon[1];
+
+        Task newTask;
+
+        if (taskType.equals("T")) {
+            newTask = new Todo(taskDescription);
+        } else if (taskType.equals("E")) {
+            String[] toGetFromTo = taskDescription.split(" \\(from: | to: |\\)");
+            String task = toGetFromTo[0];
+            LocalDateTime from = convertToDateTime(toGetFromTo[1]);
+            LocalDateTime to = convertToDateTime(toGetFromTo[2]);
+            newTask = new Event(task, from, to);
+
```

Comment:
Perhaps you can remove this whitespace, as well as the one in the else block below?